Makefile

```diff
@@ -1,28 +1,27 @@
-vpf-gpu:
-	DOCKER_BUILDKIT=1 docker build -f docker/Dockerfile.gpu --tag vpf-gpu .
+vpf-docker:
+	DOCKER_BUILDKIT=1 docker build -f docker/Dockerfile --tag vpf-gpu \
+					--build-arg GEN_PYTORCH_EXT=1 .
+					.
 
-vpf-gpu-all:
+vpf-docker-tensorrt:
 	DOCKER_BUILDKIT=1 docker build \
-					--tag vpf-gpu-all \
-					-f docker/Dockerfile.gpu \
-					--build-arg GEN_PYTORCH_EXT=1 \
-					--build-arg GEN_OPENGL_EXT=1 \
+					--tag vpf-gpu-trt \
+					-f docker/Dockerfile.tensorrt \
 					.
 
-run_tests: vpf-gpu
-	docker run --rm -v $(shell pwd):/repo --entrypoint "python3" vpf-gpu  -m unittest discover /repo/tests /repo/tests/test.mp4
+run_tests: vpf-docker
+	docker run --gpus all --rm -v $(shell pwd):/repo --entrypoint "python3" vpf-gpu  -m unittest discover /repo/tests
 
-run_samples: vpf-gpu-all
-	docker run --rm -v $(shell pwd):/repo --entrypoint "python3" vpf-gpu-all -m unittest discover /repo/tests /repo/tests/test.mp4
-	docker run --rm -v $(shell pwd):/repo --workdir /repo --entrypoint "make" vpf-gpu-all run_samples_without_docker
+run_samples: vpf-docker-tensorrt
+	docker run --gpus all --rm -v $(shell pwd):/repo --workdir /repo --entrypoint "make" vpf-gpu-trt run_samples_without_docker
 
 run_samples_without_docker:
 	wget http://www.scikit-video.org/stable/_static/bikes.mp4
-
-	python ./samples/SampleDecode.py -g 0 -e ./bikes.mp4 -r ./tests/test.raw       
-	python ./samples/SampleDecodeSw.py -e ./bikes.mp4 -r ./tests/test.raw
-	python ./samples/SampleEncodeMultiThread.py 0 848 464 ./tests/test.raw 10
-	python ./samples/SampleMeasureVideoQuality.py -g 0 -i ./tests/test.raw -o ./tests/test.raw -w 848 -h 464   
+	python3 -m pip install -r ./samples/requirements.txt
```

Comment:
could the samples be properly integrated into the tests? and we get ride of the Makefile? Or too much effort?

Reply:
@rnaskulwar actually created a list of tests that he ran manually. We could download the contents in the test files

Reply:
The make file annoys me as well but i was too lazy to search for all references of it. Sure we can make it obsolete.